Trading/Mode/market_maker_trading_mode.py

```diff
@@ -0,0 +1,206 @@
+        self.fees = {}
+
+        if MarketMakerTradingMode.DELTA_ASK not in trading_mode.trading_mode_config or \
+                MarketMakerTradingMode.DELTA_BID not in trading_mode.trading_mode_config:
+            self.logger.error(f"Can't create any trade : some configuration is missing "
+                              f"in {MarketMakerTradingMode.get_config_file_name()}, "
+                              f"please check {MarketMakerTradingMode.DELTA_ASK} and {MarketMakerTradingMode.DELTA_BID}")
+        else:
+            self.config_delta_ask = trading_mode.trading_mode_config[MarketMakerTradingMode.DELTA_ASK]
+            self.config_delta_bid = trading_mode.trading_mode_config[MarketMakerTradingMode.DELTA_BID]
+
+    def verify_and_adapt_delta_with_fees(self, symbol):
+        if symbol in self.fees:
+            return self.fees[symbol]
+
+        exchange_fees = self.trading_mode.exchange.get_fees(symbol)
+        delta_ask = self.config_delta_ask
+        delta_bid = self.config_delta_bid
+
+        # check ask -> limit_orders -> MAKER ? not sure -> max
+        common_fees = max(exchange_fees[ExchangeConstantsMarketPropertyColumns.TAKER.value],
+                          exchange_fees[ExchangeConstantsMarketPropertyColumns.FEE.value],
+                          exchange_fees[ExchangeConstantsMarketPropertyColumns.MAKER.value])
+
+        if delta_ask < (common_fees / self.FEES_ATTENUATION):
+            delta_ask = common_fees / self.FEES_ATTENUATION
+
+        if delta_bid < (common_fees / self.FEES_ATTENUATION):
+            delta_bid = common_fees / self.FEES_ATTENUATION
+
+        self.fees[symbol] = delta_ask, delta_bid
+        return self.fees[symbol]
+
+    def _get_quantity_from_risk(self, trader, quantity):
+        return quantity * trader.get_risk() / self.LIMIT_ORDER_ATTENUATION
+
+    @staticmethod
+    async def can_create_order(symbol, exchange, state, portfolio):
+        return True
+
+    async def create_new_order(self, eval_note, symbol, exchange, trader, portfolio, state):
+        current_order = None
+
+        try:
+            delta_ask, delta_bid = self.verify_and_adapt_delta_with_fees(symbol)
+            best_bid_price = eval_note[ExchangeConstantsOrderBookInfoColumns.BIDS.value][0]
+            best_ask_price = eval_note[ExchangeConstantsOrderBookInfoColumns.ASKS.value][0]
+
+            current_symbol_holding, current_market_quantity, market_quantity, price, symbol_market = \
+                await self.get_pre_order_data(exchange, symbol, portfolio)
+
+            created_orders = []
+
+            # Create SHORT order
+            quantity = self._get_quantity_from_risk(trader, current_symbol_holding)
+            quantity = self.add_dusts_to_quantity_if_necessary(quantity, price,
+                                                               symbol_market, current_symbol_holding)
+            limit_price = best_ask_price + (best_ask_price * delta_ask * quantity)
```

Comment:
Just for info, why (in short and long) add a linear correlation to quantity ? If I understand it well, it will make orders further and further from the current price as quantity increase and with enough quantity it will never trade anything because too far in the order book ?

Reply:
Yes, you are totally right.